src/frontEnd/nav/onSideNavCollapseRequested.ts

```diff
@@ -1,4 +1,16 @@
-export const onSideNavCollapseRequested = (state) => ({
-    ...state,
-    isSideNavOpen: false
-});
+import { navTo } from "./navTo"
+
+export const onSideNavCollapseRequested = (state, destination) => {
+    console.log(destination);
+    const newState =  {
+        ...state,
+        isSideNavOpen: false
+    }
+
+    const action = destination
```

Comment:
hmm, i would think that destination is always defined with a value by the time it hits this block

Reply:
there is one situation when the side nav needs to be closed, but no navTo needs to be performed. In this situation the call to `onSideNavCollapseRequested` is passed undefined

(ala `SideNav.tsx` - line 27)

Reply:
also, we only want to force side nav to close if mobile, so here is another place where you will call the new `isMobile()` method that you're gonna DRY up and put in its own file; if not mobile, we want to leave the sidenav as is.

Reply:
okay, gotcha!

Reply:
done and done